### ISSUE TITLE
EPUB landmarks UI (for list of illustrations, figures, etc.)

### ISSUE THREAD

**Issue by danielweck** [readium/readium-desktop#81]
This is implemented in `r2-testapp-js`, but is currently missing in `readium-desktop`.
Note that there may be multiple types of landmarks, so this is more like a set of categorized lists.

**Comment by danielweck** [comment]
Related: https://github.com/edrlab/readium-desktop/issues/82

**Comment by danielweck** [comment]
Perhaps use a tab user interface? 
https://github.com/reactjs/react-tabs

**Comment by clebeaupin** [comment]
Check this issue on children literature epubs

**Comment by danielweck** [comment]
This does not seem to be implemented correctly, see:
https://github.com/readium/readium-desktop/issues/369

This is how it looks in `r2-testapp-js`:
![R2-testapp-Landmarks](https://user-images.githubusercontent.com/625201/60002335-5b0b3a00-9660-11e9-9cc4-46e4f42be1ac.png)

**Comment by danielweck** [comment]
Related issue: https://github.com/readium/readium-desktop/issues/400

**Comment by danielweck** [comment]
The current Thorium UI only exposes "list of illustrations" (see screenshot below).
This is not how EPUB landmarks work, see screenshot above (r2-testapp-js) which shows the "Main" landmarks defined in the EPUB3 Navigation Document of Children's Litterature ( https://github.com/IDPF/epub3-samples/blob/0b3e80553f509fbdfe591008cd7d1e804b24db54/30/childrens-literature/EPUB/nav.xhtml#L139-L145 )

![ThoriumPageList](https://user-images.githubusercontent.com/625201/63063264-df0cd000-bef3-11e9-894a-627be1e2a602.png)

**Comment by danielweck** [comment]
Other landmarks examples:
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/moby-dick-mo/OPS/toc.xhtml#L441-L454
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/cc-shared-culture/EPUB/xhtml/toc.xhtml#L59-L69
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/indexing-for-eds-and-auths-3f/EPUB/ncx.xhtml#L33-L85
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/linear-algebra/EPUB/xhtml/nav.xhtml#L323-L329
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/epub30-spec/EPUB/xhtml/epub30-nav.xhtml#L552-L560
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/regime-anticancer-arabic/EPUB/Navigation/nav.xhtml#L29-L40
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/sous-le-vent_svg-in-spine/EPUB/Navigation/nav.xhtml#L20-L28
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/GhV-oeb-page/EPUB/Navigation/nav.xhtml#L24-L35
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/kusamakura-preview/EPUB/xhtml/%E7%9B%AE%E6%AC%A1.xhtml#L61-L74
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/wasteland-otf-obf/EPUB/wasteland-nav.xhtml#L20-L29

**Comment by danielweck** [comment]
LOI (List Of Illustrations) is a single link to an HTML list somewhere in a publication document, see:
https://www.idpf.org/epub/30/spec/epub30-contentdocs.html#sec-xhtml-nav-def-types-landmarks

**Comment by llemeurfr** [comment]
That's right, we have still not implemented a landmarks "tab". 

We should perhaps remove the "Table of Illustrations" entry for now and prepare a proper UI for full landmarks.

**Comment by llemeurfr** [comment]
Landmarks navigation enables quick access to major sections of the content. They are somewhat similar to a simplified ToC, and therefore complementary.

I propose to modify the "navigation" panel as follows:
- suppress Table of Illustrations (always greyed).
- If landmarks are present in the document (<nav epub:type="landmarks"&gt;), display "Landmarks" ("Repères" in fr, conserver Landmarks in German for now) + the list of links below as a SideMenuSection.

**Comment by llemeurfr** [comment]
Also, in the code, some variables are called "landmark" when they are not. Let's rename such variables so that we maintain a good & clear codebase.

**Comment by danielweck** [comment]
Copy-pasting information from https://github.com/readium/readium-desktop/issues/400

____

Issue spawned from: https://github.com/readium/readium-desktop/pull/380#issuecomment-508864841

https://github.com/readium/readium-desktop/blob/23a0b6438f78e3096f6fb808cf46a64127e771f2/src/renderer/components/reader/ReaderMenu.tsx#L80-L84

https://github.com/readium/readium-desktop/blob/23a0b6438f78e3096f6fb808cf46a64127e771f2/src/renderer/components/reader/ReaderMenu.tsx#L156-L192

Related issue: https://github.com/readium/readium-desktop/issues/369

Looking at the current UI, I think there was some confusion about EPUB's 3 major types of *authored* navigation structures:

* table of contents (TOC)
* page list (authored list of `epub:type` page-breaks)
* landmarks (which can be `epub:type` list-of-illustration, etc.)

Then, reading systems add their own "bookmarks" and "annotations" lists, which are *not* authored in the original publications, but added as navigation overlay.

**Comment by Scarsniik** [comment]
Fixed by #653
